src/client/datascience/jupyter/kernels/kernelExecution.ts

```diff
@@ -183,8 +183,8 @@ export class KernelExecution implements IDisposable {
         const restarted = createDeferred<boolean>();
         const stopWatch = new StopWatch();
         // Listen to status change events so we can tell if we're restarting
-        const restartHandler = (e: ServerStatus) => {
-            if (e === ServerStatus.Restarting) {
+        const restartHandler = (e: KernelMessage.Status) => {
+            if (e === 'restarting' || e === 'autorestarting') {
```

Comment:
Interesting didn't realize there were two restarting states.

Reply:
FOund that recently, they differentiate between regular restart and automatic restarts.
I think this was added specifically for displaying a popup letting the user know the kernel died & its automatically being restarted.